diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5006f78..3a559bf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -6,11 +6,11 @@ project(tdlib-purple VERSION ${VERSION} LANGUAGES CXX)
 
 find_package(Td REQUIRED)
 math(EXPR TDLIB_VERSION_NUMBER "10000*${Td_VERSION_MAJOR} + 100*${Td_VERSION_MINOR} + ${Td_VERSION_PATCH}")
-if (NOT(${TDLIB_VERSION_NUMBER} EQUAL 10709))
+if (${TDLIB_VERSION_NUMBER} LESS 10800)
     # Error message must begin with "tdlib version" for a grep command from readme
-    message(FATAL_ERROR "tdlib version 1.7.9 is required
+    message(FATAL_ERROR "at least tdlib version 1.8.0 is required
 (version found: ${Td_VERSION_MAJOR}.${Td_VERSION_MINOR}.${Td_VERSION_PATCH})")
-endif (NOT(${TDLIB_VERSION_NUMBER} EQUAL 10709))
+endif (${TDLIB_VERSION_NUMBER} LESS 10800)
 
 set(NoPkgConfig FALSE CACHE BOOL "Do not use pkg-config")
 set(NoWebp FALSE CACHE BOOL "Do not decode webp stickers")
@@ -18,8 +18,6 @@ set(NoBundledLottie FALSE CACHE BOOL "Do not use bundled rlottie library")
 set(NoLottie FALSE CACHE BOOL "Disable animated sticker conversion")
 set(NoTranslations FALSE CACHE BOOL "Disable translation support")
 set(NoVoip FALSE CACHE BOOL "Disable voice call support")
-set(tgvoip_INCLUDE_DIRS "" CACHE STRING "Path to libtgvoip headers")
-set(tgvoip_LIBRARIES "tgvoip" CACHE STRING "tgvoip library to link against")
 set(API_ID 94575 CACHE STRING "API id")
 set(API_HASH a3406de8d171bb422bb6ddf3bbd800e2 CACHE STRING "API hash")
 set(STUFF "" CACHE STRING "")
@@ -31,6 +29,10 @@ if (NOT NoPkgConfig)
         pkg_check_modules(libwebp libwebp)
         pkg_check_modules(libpng libpng)
     endif (NOT NoWebp)
+    if (NOT NoVoip)
+        pkg_check_modules(tgvoip tgvoip)
+    endif(NOT NoVoip)
+
     pkg_get_variable(PURPLE_PLUGIN_DIR purple plugindir)
     pkg_get_variable(PURPLE_DATA_DIR purple datarootdir)
 endif (NOT NoPkgConfig)
